File: Android/Sources/UserControlLibraries/GaodeMaps/src/main/java/com/genexus/android/controls/maps/gaode/MapLocationBounds.java
package com.genexus.android.controls.maps.gaode;

import com.amap.api.maps.model.LatLngBounds;
import com.genexus.android.core.controls.maps.common.IMapLocationBounds;

class MapLocationBounds implements IMapLocationBounds<MapLocation> {
	private final LatLngBounds mBounds;

	public MapLocationBounds(LatLngBounds bounds) {
		mBounds = bounds;
	}

	LatLngBounds getLatLngBounds() {
		return mBounds;
	}

	@Override
	public MapLocation southwest() {
		return new MapLocation(mBounds.southwest);
	}

	@Override
	public MapLocation northeast() {
		return new MapLocation(mBounds.northeast);
	}
}

File: Android/Sources/UserControlLibraries/MapboxLib/src/main/java/com/genexus/android/controls/maps/mapbox/MapLocationBounds.kt
package com.genexus.android.controls.maps.mapbox

import com.genexus.android.core.controls.maps.common.IMapLocationBounds
import com.mapbox.mapboxsdk.geometry.LatLngBounds

class MapLocationBounds(val latLngBounds: LatLngBounds) : IMapLocationBounds<MapLocation> {

    override fun southwest(): MapLocation {
        return MapLocation(latLngBounds.southWest)
    }

    override fun northeast(): MapLocation {
        return MapLocation(latLngBounds.northEast)
    }
}
